packages/vitest-preview/src/preview.ts

```diff
@@ -42,3 +44,88 @@ export function debug(): void {
     document.documentElement.outerHTML,
   );
 }
+
+/**
+ * Watches for changes in the document and automatically calls debug() when changes occur.
+ * Uses MutationObserver to detect DOM mutations.
+ *
+ * @param options - Configuration options for the observer
+ * @param options.throttle - Time in milliseconds to throttle debug calls (default: 50). Set to null to disable throttling.
+ * @param options.start - Whether to call debug immediately when watch is started (default: true)
+ * @param options.end - Whether to call debug one final time before stopping watching (default: true)
+ * @param options.debug - Whether to log the total number of debug calls to the console (default: false)
+ * @returns A function to stop watching
+ */
+export function watch(
+  options: {
+    throttle?: number | null;
+    start?: boolean;
+    end?: boolean;
+    debug?: boolean;
+  } = {},
+): () => void {
+  // Stop any existing observer
+  if (observer) {
+    observer.disconnect();
+    observer = null;
+  }
+
+  // Set up throttling to avoid too many debug() calls
+  const throttleTime = options.throttle ?? 50;
+  const start = options.start ?? true;
+  const end = options.end ?? true;
+  const debugFlag = options.debug ?? false;
+  let timeout: ReturnType<typeof setTimeout> | null = null;
+
+  let totalCalls = 0;
+  if (start) {
+    debug();
+    totalCalls++;
+  }
+
+  let lastCall = 0;
+  // Create a new observer
+  observer = new MutationObserver(() => {
+    // Throttle debug calls
+    if (throttleTime === null) {
+      debug();
+      totalCalls++;
+      return;
+    } else {
```

Comment:
This condition seems unnecessary since `throttleTime` will never be `null`.

Reply:
@ntt261298 User can set `throttle` to `null` by 
```
watch({
    throttle:  null, // Does not throttle
  });
```

Reply:
Oh I see the issue with nullish coalescing. If I pass `thottle` `null` it will also be `50`

I pushed a change. Thanks